Mathlib/Algebra/Module/LocalizedModule.lean

```diff
@@ -1082,6 +1085,82 @@ theorem mkOfAlgebra {R S S' : Type _} [CommRing R] [CommRing S] [CommRing S'] [A
 
 end Algebra
 
+variable [Module (Localization S) M'] [IsScalarTower R (Localization S) M']
+
+/-- If `(f : M →ₗ[R] M')` is a localization of modules, then the map
+`(localization S) × M → N, (s, m) ↦ s • f m` is the tensor product (insomuch as it is the universal
+bilinear map).
+In particular, there is an isomorphism between `localized_module S M` and `(localization S) ⊗[R] M`
+given by `m/s ↦ (1/s) ⊗ₜ m`.
+--/
+theorem isBaseChange : IsBaseChange (Localization S) f := by
+  apply IsBaseChange.of_lift_unique
+  intros Q _ _ _ _ g
+  have := IsLocalizedModule.is_universal S f g <| by
+      intro s
+      rw [Module.End_isUnit_iff, Function.bijective_iff_existsUnique]
+      intro q
+      use Localization.mk 1 s • q
+      dsimp only
+      constructor
+      on_goal 1 => rw [Module.algebraMap_end_apply]
+      on_goal 2 =>
+        intro q' h
+        rw [Module.algebraMap_end_apply] at h
+        rw [← h, smul_comm]
+      all_goals
+        rw [← smul_assoc, Localization.smul_mk, smul_eq_mul, mul_one, Localization.mk_self,
+          one_smul]
+  rcases this with ⟨ℓ, h₁, h₂⟩
+  -- Should this be refactored into a `linear_map.extend_scalars` lemma? If so, what would the
+  -- predicate have to be? Just `map_smul'`?
+  let g' : M' →ₗ[Localization S] Q :=
+    { toFun := ℓ.toFun
+      map_add' := ℓ.map_add'
+      map_smul' := by
+        intros r x
+        rw [RingHom.id_apply]
+        induction' r using Localization.induction_on with data
+        rcases data with ⟨r, s⟩
+        conv_lhs =>
+          rw [← one_smul (Localization S) (ℓ.toFun _), ← Localization.mk_self s, ← mul_one r, ←
+            @mul_one R _ ↑s]
+          conv =>
+            pattern (occs := 1 2) Localization.mk _ s <;>
+              (rw [← smul_eq_mul R, ← Localization.smul_mk])
+          conv =>
+            pattern (r • _) • _
+            rw [smul_assoc]
+          rw [ℓ.map_smul', RingHom.id_apply]
+          conv =>
+            pattern (↑s • _) • _
+            rw [smul_comm]
+          conv =>
+            pattern (↑s • _) • _
```

Comment:
Just mentioning to whoever is reviewing that I created #5786 so that this can be written instead as
```suggestion
          conv in (↑s • _) • _ =>
```

The way `pattern` handles `occs` is a little weird in the context of `in` clauses, so I'll also mention that here's a way you could write the one above that uses it:
```lean
          conv in (occs := 1 2) Localization.mk _ s => all_goals
              rw [← smul_eq_mul R, ← Localization.smul_mk]
```

Reply:
thanks! done.